### PR TITLE
refactor: switch to clap

### LINKED ISSUE #1009: Migrate CLI from StructOpt to Clap
Seems like [StructOpt](https://github.com/TeXitoi/structopt#maintenance) is now in maintenance mode and the recommendation is to migrate to [Clap](https://github.com/clap-rs/clap).

Also, this might be a good opportunity to refactor some of the commands to take required parameters directly (rather than as flags). For example:
```
miden run example.masm
```
Instead of:
```
miden run -a example.masm
```

### PR DESCRIPTION
**Pull request by leviathanbeak**
## Describe your changes
Switches from `structopt` to `clap` library for CLI app.
Closes #1009 

## Checklist before requesting a review
- Repo forked and branch created from `main` according to naming convention.
- Commit messages and codestyle follow [conventions](./CONTRIBUTING.md).
- Relevant issues are linked in the PR description.
- Documentation/comments updated according to changes.

### CODE REVIEW

File: miden/src/main.rs [hunk 1]
```diff
@@ -49,7 +48,7 @@ impl Cli {
 /// Executable entry point
 pub fn main() {
     // read command-line args
-    let cli = Cli::from_args();
+    let cli = Cli::try_parse().unwrap();
```
Comment (bobbinth) [inline miden/src/main.rs:51]: Should this be `let cli = Cli::parse();`? This way, we'd get a pretty-printed message if arguments were not supplied correctly.

**Review by bobbinth — COMMENTED** [review]
Looks great! Thank you for doing this. I left one small comment inline - but wanted to mention one other thing: we do active development in the `next` branch - and this is where the most up-to-date code is. Could you rebase this from `next` (this will also fix clippy errors)?

**Comment by leviathanbeak** [comment]
got it, will rebase tomorrow, thanks

**Comment by leviathanbeak** [comment]
I forgot about this, closing in favor of #1044

**Pull request closed without merging**